src/xds/client.rs

```diff
@@ -274,16 +276,27 @@ impl AdsClient {
             .into_inner();
 
         info!("Stream established");
+        let (tx, initial_xds_rx) = oneshot::channel();
+        let mut initial_xds_tx = Some(tx);
+        let ready = mem::take(&mut self.block_ready);
```

Comment:
I think since we take it, if this function exits unsuccessfully we will still end up dropping it (?)

Reply:
Ah, good call. So `ready` gets `move`d into the closure passed to `tokio::spawn`, so it'll `drop` when that closure exits. The only way for the closure to exit without a successful send is for the `Sender` to drop from scope. I'll change the code a bit so that the `tx` is within an infinite loop, guaranteeing that the only time the sender drops is after a message is passed.